tests/PySys/software-management-end-to-end/sm_mapper_fail_and_reconnect/run.py

```diff
@@ -0,0 +1,142 @@
+import sys
+import time
+import subprocess
+from pathlib import Path
+
+from pysys.basetest import BaseTest
+
+"""
+Validate the tedge-mapper-sm-c8y does not loose last message from tedge-agent when it fails and comes back
+
+Given a configured system
+When `rolldice` package is installed
+when a subscriber is started as `sudo tedge mqtt sub 'c8y/s/us'`
+When tedge_agent is started as `sudo systemctl start tedge-agent.service`
+When sm mapper is started as `sudo systemctl start tedge-mapper-sm-c8y.service`
+When send a delete operation `sudo tedge mqtt pub "c8y/s/ds" "528,tedge,rolldice,,,delete"`
+When sm mapper is stopped `sudo systemctl stop tedge-mapper-sm-c8y.service`
+Wait for sometime for operation to be completed
+When sm mapper is restarted `sudo systemctl restart tedge-mapper-sm-c8y.service`
+Now sm mapper receives the last update result message, process and forward it to the cloud on `c8y/s/us`
+Then validate subscriber output for `501,c8y_SoftwareUpdate`, for the status of operation
+Then validate subscriber output for `503,c8y_SoftwareUpdate` for final result of operation
+Then test has passed
+"""
+
+class SmMapperC8yReceiveLastMessageOnRestart(BaseTest):
+    systemctl = "/usr/bin/systemctl"
```

Comment:
On my system, `systemctl` is `/bin/systemctl`. This kind of hardcoded path makes the tests inconvenient to run.